statistics/handle/bootstrap.go

```diff
@@ -189,31 +189,49 @@ func (h *Handle) initStatsHistograms(is infoschema.InfoSchema, cache *statsCache
 
 func (h *Handle) initStatsTopN4Chunk(cache *statsCache, iter *chunk.Iterator4Chunk) {
 	affectedIndexes := make(map[*statistics.Index]struct{})
+	affectedColumns := make(map[*statistics.Column]struct{})
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
 		table, ok := cache.Get(row.GetInt64(0))
 		if !ok {
 			continue
 		}
-		idx, ok := table.Indices[row.GetInt64(1)]
-		if !ok || (idx.CMSketch == nil && idx.StatsVer <= statistics.Version1) {
-			continue
-		}
-		if idx.TopN == nil {
-			idx.TopN = statistics.NewTopN(32)
+		if row.GetInt64(1) > 0 {
+			idx, ok := table.Indices[row.GetInt64(2)]
+			if !ok || (idx.CMSketch == nil && idx.StatsVer <= statistics.Version1) {
+				continue
+			}
+			if idx.TopN == nil {
+				idx.TopN = statistics.NewTopN(32)
+			}
+			affectedIndexes[idx] = struct{}{}
+			data := make([]byte, len(row.GetBytes(3)))
+			copy(data, row.GetBytes(3))
+			idx.TopN.AppendTopN(data, row.GetUint64(4))
+		} else {
+			col, ok := table.Columns[row.GetInt64(2)]
+			if !ok || col.Info == nil || !mysql.HasPriKeyFlag(col.Info.GetFlag()) || (col.CMSketch == nil && col.StatsVer <= statistics.Version1) {
```

Comment:
Looks like this fix should also apply to stats ver1🤔

Reply:
I did some simple experiments and believe this bug also affects stats ver1, so probably this fix should also apply to stats ver1.

Reply:
Got it. I will make the fix apply to stats ver1.

Reply:
The condititon `col.CMSketch == nil && col.StatsVer <= statistics.Version1` is copied from the condition of loading index TopN(`idx.CMSketch == nil && idx.StatsVer <= statistics.Version1`), which I don't fully understand by now. The condition is from #14421 and #24623. I will take a deep look into it.